src/libtriton/api/api.cpp

```diff
@@ -367,33 +366,38 @@ namespace triton {
   void API::initEngines(void) {
     this->checkArchitecture();
 
-    this->modes = new(std::nothrow) triton::modes::Modes();
-    if (this->modes == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+    do{
+      this->modes = new(std::nothrow) triton::modes::Modes();
+      if (this->modes == nullptr)
+        break;
+
+      this->symbolic = new(std::nothrow) triton::engines::symbolic::SymbolicEngine(&this->arch, this->modes, &this->callbacks);
+      if (this->symbolic == nullptr)
+        break;
 
-    this->symbolic = new(std::nothrow) triton::engines::symbolic::SymbolicEngine(&this->arch, this->modes, &this->callbacks);
-    if (this->symbolic == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      this->solver = new(std::nothrow) triton::engines::solver::SolverEngine(this->symbolic);
+      if (this->solver == nullptr)
+        break;
 
-    this->solver = new(std::nothrow) triton::engines::solver::SolverEngine(this->symbolic);
-    if (this->solver == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      this->astGarbageCollector = new(std::nothrow) triton::ast::AstGarbageCollector(this->modes);
+      if (this->astGarbageCollector == nullptr)
+        break;
 
-    this->astGarbageCollector = new(std::nothrow) triton::ast::AstGarbageCollector(this->modes);
-    if (this->astGarbageCollector == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      this->taint = new(std::nothrow) triton::engines::taint::TaintEngine(this->symbolic);
+      if (this->taint == nullptr)
+        break;
 
-    this->taint = new(std::nothrow) triton::engines::taint::TaintEngine(this->symbolic);
-    if (this->taint == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      this->irBuilder = new(std::nothrow) triton::arch::IrBuilder(&this->arch, this->modes, this->astGarbageCollector, this->symbolic, this->taint);
+      if (this->irBuilder == nullptr)
+        break;
 
-    this->irBuilder = new(std::nothrow) triton::arch::IrBuilder(&this->arch, this->modes, this->astGarbageCollector, this->symbolic, this->taint);
-    if (this->irBuilder == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      this->z3Interface = new(std::nothrow) triton::ast::Z3Interface(this->symbolic);
+      if (this->z3Interface == nullptr)
+        break;
 
-    this->z3Interface = new(std::nothrow) triton::ast::Z3Interface(this->symbolic);
-    if (this->z3Interface == nullptr)
-      throw triton::exceptions::API("API::initEngines(): No enough memory.");
+      return;
+    }while(0);
```

Comment:
In VS2015, use MinSize build option, the assembly code of the initEngines() was less 247 bytes than old. The old code were 757 bytes, the new were 510 bytes, there was 32% saved. In linux src, many MACRO were do-while style. If you don't really like it, I'll revert it.